woocommerce/woocommerce-gateway-stripe
# Fix Stripe payment method ordering

## Changes proposed in this Pull Request:
Issue: Currently, when the `woocommerce_gateway_order` option does not yet have the `stripe` key, any custom ordering of Stripe payment methods gets ignored, i.e. the custom ordering in admin does not match checkout. 

For example, on a fresh install, the `woocommerce_gateway_order` option does not yet exist, even when Stripe and other payment gateways are enabled, so any custom ordering the merchant makes in the Stripe payment methods page gets ignored.

This PR adds logic to ensure payment method ordering gets even for such scenarios.

Fixes #3454 
 






## Screenshots

**BUG: custom ordering is ignored**
<img width="800" alt="Screenshot 2024-09-24 at 1 12 48 PM" src="https://github.com/user-attachments/assets/75e93eff-143b-43d2-a85a-3df49bc9193f">


## Testing instructions
1. Simulate a fresh install by deleting `woocommerce_gateway_order` from the `wp_options` table.
    - `DELETE FROM wp_otions where option_name = 'woocommerce_gateway_order';` 
2. Go to the Stripe payment methods page, enable a few payment methods and re-arrange them. Save your changes.
    - `wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=methods`
3. As a shopper, add an item to cart and go to checkout.
4. Verify the ordering of the payment methods is correct.
5. Also test the ordering of Stripe against other gateways, e.g. `Cash on delivery`, `Check payments`
    - `wp-admin/admin.php?page=wc-settings&tab=checkout`






---


**Post merge**
